### PR TITLE
[RSDK-8422] add make commands for building installer

### PR DESCRIPTION
**Pull request by gvaradarajan**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -36,6 +36,12 @@ build:
 build-native:
 	cargo build -p micro-rdk-server --bin micro-rdk-server-native
 
+build-installer:
+	cargo build -p micro-rdk-installer --bin micro-rdk-installer --release
+
+install-tools:
+	cargo install cargo-generate espflash espup cargo-espflash ldproxy; espup install -s -f ~/esp/export-rs.sh -v 1.67.0
```
[the remark this answers is not in the document]
Comment (mattjperez) [reply Makefile:43]: In `make build-esp32-bin` we use `cargo espflash save-image` to compile a single binary that contains the partition table and program. A single artifact we can publish on GitHub.

Strangely, it doesn't appear that `cargo-espflash` can actually flash this single binary that it wrote. [Here'](https://github.com/esp-rs/espflash/issues/364)s an example of other users getting confused. Note that they too end up using `espflash`, though they actually point to the elf file in the `target/release` subfolder.

In `make flash-esp32-bin`, we are using `espflash write-bin` to write the complete binary to the device from `0x0`. `write-bin` lets us write an arbitrary binary to an offset, not requiring an ELF file the way `espflash flash` does.

We actually use `write-bin` under the hood via the `espflash` crate for `write-flash`, `update-app-image`, and `write-credentials`. This makes our installer's behavior closer the closest thing to the current workflow.

**Comment by npmenard** [comment]
I don't think we should take the burden of automating the installation of the tools it means we should also deal with updates, dependency handling, different platform etc... I believe this is more of a documentation problem where we can give general instructions while pointing to the relevant tool descriptions.
Generally devs should be using thew docker image initially until they are sufficiently frustrated to do their own  environment setup

File: Makefile [hunk 2]
```diff
@@ -36,6 +36,13 @@ build:
 build-native:
 	cargo build -p micro-rdk-server --bin micro-rdk-server-native
 
+build-installer:
+	cargo build -p micro-rdk-installer --bin micro-rdk-installer --release
+
+install-tools:
```
Comment (acmorrow) [inline Makefile:42]: It sounds like we should remove the `install-tools` target and only retain `build-installer.

**Comment by gvaradarajan** [comment]
I agree with @npmenard and @acmorrow on removing the tool installations from the Makefile, but I'd like to give @mattjperez a chance to respond with any objections as the creator of the original ticket

**Comment by mattjperez** [comment]
> I agree with @npmenard and @acmorrow on removing the tool installations from the Makefile, but I'd like to give @mattjperez a chance to respond with any objections as the creator of the original ticket

Maybe I'm not thinking of it the right way, but "automating the installation of the tools" was kinda what we did with the new esp-idf setup. leveraging rust and cargo to handle the host's dependecies,etc seems similar in how we (now) rely on embed to compile to the user's host system and install everything for building esp-idf.

I imagined the docs (since we can build locally now) for a new user eventually being more like: `install rust -> install cargo-generate -> generate project template -> make install-tooling -> make build-esp32-bin`. 

If the consensus is this doesn't make sense or add value, kill it.

**Comment by gvaradarajan** [comment]
I think adding clarifications to docs for the idiosyncrasies that can be present in various systems when trying to install tooling is easier than trying to account for all scenarios in a make command, so I'm proceeding to remove

**Pull request merged**